be/src/exec/pipeline/scan_operator.cpp

```diff
@@ -176,25 +176,29 @@ Status ScanOperator::_trigger_next_scan(RuntimeState* state, int chunk_source_in
     _is_io_task_running[chunk_source_index] = true;
 
     bool offer_task_success = false;
+    std::weak_ptr<ScanOperator> wp = shared_from_this();
     if (_workgroup != nullptr) {
-        workgroup::ScanTask task = workgroup::ScanTask(_workgroup, [this, state, chunk_source_index](int worker_id) {
-            {
-                SCOPED_THREAD_LOCAL_MEM_TRACKER_SETTER(state->instance_mem_tracker());
-                size_t num_read_chunks = 0;
-                _chunk_sources[chunk_source_index]->buffer_next_batch_chunks_blocking_for_workgroup(
-                        _buffer_size, _is_finished, &num_read_chunks, worker_id, _workgroup);
-                // TODO (by laotan332): More detailed information is needed
-                _workgroup->incr_period_scaned_chunk_num(num_read_chunks);
-                _workgroup->increment_real_runtime_ns(_chunk_sources[chunk_source_index]->last_spent_cpu_time_ns());
-
-                // for big query check
-                COUNTER_UPDATE(_total_cost_cpu_time_ns_counter,
-                               _chunk_sources[chunk_source_index]->last_spent_cpu_time_ns());
-                _last_scan_rows_num += _chunk_sources[chunk_source_index]->last_scan_rows_num();
+        workgroup::ScanTask task = workgroup::ScanTask(_workgroup, [wp, state, chunk_source_index](int worker_id) {
+            if (auto sp = wp.lock()) {
```

Comment:
The i/o task also uses some other fragment-shared resources, such as `RuntimState` and `OlapScanNode`, which may be already deconstructed, if the query is finished and fragment context is deconstructed.

Reply:
If we ensure that `ScanOperator` is released before `RuntimState` and `OlapScanNode` in `~FragmentContext()`, maybe we could ensure this resources not deconstructed if `sp != nullptr`.

Reply:
> If we ensure that `ScanOperator` is released before `RuntimState` and `OlapScanNode` in `~FragmentContext()`, maybe we could ensure this resources not deconstructed if `sp != nullptr`.

we have this guarantess now, you can look at the order in which they are defined in `FragmentContext`.